index.js

```diff
@@ -10,7 +10,13 @@ module.exports = function kindOf(val) {
   if (type === 'number') return 'number';
   if (type === 'symbol') return 'symbol';
   if (type === 'function') {
-    return isGeneratorFn(val) ? 'generatorfunction' : 'function';
+    if (isGeneratorFn(val)) {
+      return 'generatorfunction'
+    } else if (isAsyncFn(val)) {
```

Comment:
no need for `else if` here since the above condition already returns if `true`. Same with the next condition.